### PR TITLE
[tests-only][full-ci]Check for suite existence in the expected to failure files

### PR DESCRIPTION
**Pull request by SagarGi**
### Description
This PR adds a pipeline that checks whether there are the tests scenarios in the expected failure files that is in the tests suites inside `/features`.
### Related Issue:
https://github.com/owncloud/web/issues/9778

### CODE REVIEW

**Comment by SagarGi** [comment]
This is how the script works: https://drone.owncloud.com/owncloud/web/39580/3/2

File: .drone.star [hunk 1]
```diff
@@ -1681,6 +1682,34 @@ def licenseCheck(ctx):
         },
     }]
 
+def checkTestSuitesInExpectedToFailure(ctx):
```
Comment (saw-jan) [inline .drone.star:1685]:
```suggestion
def checkTestSuitesInExpectedFailures(ctx):
```

File: .drone.star [hunk 2]
```diff
@@ -1681,6 +1682,34 @@ def licenseCheck(ctx):
...
+    return [{
+        "kind": "pipeline",
+        "type": "docker",
+        "name": "check-suite-in-expected-to-failure",
```
Comment (saw-jan) [inline .drone.star:1689]:
```suggestion
        "name": "check-suites-in-expected-failures",
```

File: .drone.star [hunk 3]
```diff
@@ -1681,6 +1682,34 @@ def licenseCheck(ctx):
...
+        "workspace": {
+            "base": dir["base"],
+            "path": config["app"],
+        },
+        "steps": [
+            {
+                "name": "check-suites",
+                "image": OC_CI_ALPINE,
+                "commands": [
+                    "cd %s/tests/acceptance && ./check-deleted-suites-in-expected-to-failure.sh" % dir["web"],
```
Comment (saw-jan) [inline .drone.star:1699]:
```suggestion
                    "%s/tests/acceptance/check-deleted-suites-in-expected-to-failure.sh" % dir["web"],
```

File: .drone.star [hunk 4]
```diff
@@ -1681,6 +1682,34 @@ def licenseCheck(ctx):
...
+                ],
+            },
+        ],
+        "trigger": {
+            "ref": [
+                "refs/heads/master",
+                "refs/heads/stable-*",
+                "refs/tags/**",
```
Comment (saw-jan) [inline .drone.star:1707]: maybe we should run it in PRs only
```suggestion
```

File: tests/acceptance/check-deleted-suites-in-expected-to-failure.sh [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+
+log_success() {
+	echo -e "\e[32m$1\e[0m"
+}
+
+SCRIPT_PATH=$(dirname "$0")
+PATH_TO_SUTES="${SCRIPT_PATH}/features"
+PATH_TO_EXPECTED_TO_FAILURE_FILE="${SCRIPT_PATH}/expected-failures-with-ocis-server-ocis-storage.md"
+# contains all the suites names inside tests/acceptance/features
+SUITES_AVAILABLE=($(ls -l "$PATH_TO_SUTES" | grep '^d' | awk '{print $NF}'))
```
Comment (saw-jan) [inline tests/acceptance/check-deleted-suites-in-expected-to-failure.sh:23]:
```suggestion
AVAILABLE_SUITES=($(ls -l "$PATH_TO_SUTES" | grep '^d' | awk '{print $NF}'))
```

File: tests/acceptance/check-deleted-suites-in-expected-to-failure.sh [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+log_info() {
+	echo -e "\e[37m$1\e[0m"
+}
+
+log_success() {
+	echo -e "\e[32m$1\e[0m"
+}
+
+SCRIPT_PATH=$(dirname "$0")
+PATH_TO_SUTES="${SCRIPT_PATH}/features"
```
Comment (saw-jan) [inline tests/acceptance/check-deleted-suites-in-expected-to-failure.sh:20]:
```suggestion
PATH_TO_SUITES="${SCRIPT_PATH}/features"
```

File: tests/acceptance/check-deleted-suites-in-expected-to-failure.sh [hunk 7]
```diff
@@ -0,0 +1,63 @@
...
+PATH_TO_EXPECTED_TO_FAILURE_FILE="${SCRIPT_PATH}/expected-failures-with-ocis-server-ocis-storage.md"
+# contains all the suites names inside tests/acceptance/features
+SUITES_AVAILABLE=($(ls -l "$PATH_TO_SUTES" | grep '^d' | awk '{print $NF}'))
+
+# regex to match [someSuites/someFeatureFile.feature:lineNumber]
+REGEX_FOR_TEST_SCENARIO="\[([a-zA-Z0-9]+/[a-zA-Z0-9]+\.feature:[0-9]+)]"
```
Comment (saw-jan) [inline tests/acceptance/check-deleted-suites-in-expected-to-failure.sh:26]:
```suggestion
SCENARIO_REGEX="\[([a-zA-Z0-9]+/[a-zA-Z0-9]+\.feature:[0-9]+)]"
```

File: tests/acceptance/check-deleted-suites-in-expected-to-failure.sh [hunk 8]
```diff
@@ -0,0 +1,63 @@
...
+
+# contains all those suites available in the expected to failure files in pattern [someSuites/someFeatureFile.feature:lineNumber]
+SUITES_IN_EXPECTED_TO_FAILURE=($(grep -Eo ${REGEX_FOR_TEST_SCENARIO} ${PATH_TO_EXPECTED_TO_FAILURE_FILE}))
```
Comment (saw-jan) [inline tests/acceptance/check-deleted-suites-in-expected-to-failure.sh:29]:
```suggestion
EXPECTED_FAILURE_SCENARIOS=($(grep -Eo ${REGEX_FOR_TEST_SCENARIO} ${PATH_TO_EXPECTED_TO_FAILURE_FILE}))
```

File: tests/acceptance/check-deleted-suites-in-expected-to-failure.sh [hunk 9]
```diff
@@ -0,0 +1,63 @@
...
+
+# get and store only the suites names from SUITES_IN_EXPECTED_TO_FAILURE
+FILTERED_SUITES_IN_EXPECTED_TO_FAILURE=()
```
Comment (saw-jan) [inline tests/acceptance/check-deleted-suites-in-expected-to-failure.sh:32]:
```suggestion
EXPECTED_FAILURE_SUITES=()
```

File: tests/acceptance/check-deleted-suites-in-expected-to-failure.sh [hunk 10]
```diff
@@ -0,0 +1,63 @@
...
+for value in "${SUITES_IN_EXPECTED_TO_FAILURE[@]}"; do
+  if [[ $value =~ \[([a-zA-Z0-9]+) ]]; then
+     current_value="${BASH_REMATCH[1]}"
+     FILTERED_SUITES_IN_EXPECTED_TO_FAILURE+=("$current_value")
```
Comment (saw-jan) [inline tests/acceptance/check-deleted-suites-in-expected-to-failure.sh:36]:
```suggestion
for scenario in "${SUITES_IN_EXPECTED_TO_FAILURE[@]}"; do
  if [[ $scenario =~ \[([a-zA-Z0-9]+) ]]; then
     suite="${BASH_REMATCH[1]}"
     FILTERED_SUITES_IN_EXPECTED_TO_FAILURE+=("$suite")
```

File: tests/acceptance/check-deleted-suites-in-expected-to-failure.sh [hunk 11]
```diff
@@ -0,0 +1,63 @@
...
+  fi
+done
+
+# also filter the duplicated suites name
+FILTERED_SUITES_IN_EXPECTED_TO_FAILURE=($(echo "${FILTERED_SUITES_IN_EXPECTED_TO_FAILURE[@]}" | tr ' ' '\n' | sort | uniq))
+
+# Check the existence of the suite
+SCENARIOS_THAT_DOESNOT_EXISTS=()
```
Comment (saw-jan) [inline tests/acceptance/check-deleted-suites-in-expected-to-failure.sh:44]:
```suggestion
NONEXISTING_SUITES=()
```

File: tests/acceptance/check-deleted-suites-in-expected-to-failure.sh [hunk 12]
```diff
@@ -0,0 +1,63 @@
...
+  if [[ " ${SUITES_AVAILABLE[*]} " != *" $suite "* ]]; then
+    pattern="(${suite}/[a-zA-Z0-9]+\\.feature:[0-9]+)"
+    SCENARIOS_THAT_DOESNOT_EXISTS+=($(grep -Eo ${pattern} ${PATH_TO_EXPECTED_TO_FAILURE_FILE}))
+  fi
+done
+
+count="${#SCENARIOS_THAT_DOESNOT_EXISTS[@]}"
+
+if [ "$count" -gt 0 ]; then
+  log_info "The following test scenarios does not exists anymore:"
```
Comment (saw-jan) [inline tests/acceptance/check-deleted-suites-in-expected-to-failure.sh:55]:
```suggestion
  log_info "The following test scenarios do not exist anymore:"
```

File: tests/acceptance/check-deleted-suites-in-expected-to-failure.sh [hunk 13]
```diff
@@ -0,0 +1,63 @@
...
+  log_info "The below test scenarios can be deleted from the ${PATH_TO_EXPECTED_TO_FAILURE_FILE}."
```
Comment (saw-jan) [inline tests/acceptance/check-deleted-suites-in-expected-to-failure.sh:56]:
```suggestion
  log_info "They can be deleted from the '${PATH_TO_EXPECTED_TO_FAILURE_FILE}'."
```

File: tests/acceptance/check-deleted-suites-in-expected-to-failure.sh [hunk 14]
```diff
@@ -0,0 +1,63 @@
...
+  for path_to_scenario in "${SCENARIOS_THAT_DOESNOT_EXISTS[@]}"; do
+    log_error "$path_to_scenario"
+  done
+  exit 1
+fi
+
+log_success "All the suites in the expected to failure files exists in the test suites"
```
Comment (saw-jan) [inline tests/acceptance/check-deleted-suites-in-expected-to-failure.sh:63]:
```suggestion
log_success "All the suites in the expected failure file exist in the test suites"
```

File: tests/acceptance/check-deleted-suites-in-expected-to-failure.sh [hunk 15]
```diff
@@ -0,0 +1,63 @@
...
+  fi
+done
+
+count="${#SCENARIOS_THAT_DOESNOT_EXISTS[@]}"
+
+if [ "$count" -gt 0 ]; then
+  log_info "The following test scenarios does not exists anymore:"
+  log_info "The below test scenarios can be deleted from the ${PATH_TO_EXPECTED_TO_FAILURE_FILE}."
+  for path_to_scenario in "${SCENARIOS_THAT_DOESNOT_EXISTS[@]}"; do
+    log_error "$path_to_scenario"
```
Comment (saw-jan) [inline tests/acceptance/check-deleted-suites-in-expected-to-failure.sh:58]:
```suggestion
  for scenario_path in "${SCENARIOS_THAT_DOESNOT_EXISTS[@]}"; do
    log_error "$scenario_path"
```

File: .drone.star [hunk 16]
```diff
@@ -1681,6 +1682,31 @@ def licenseCheck(ctx):
...
+        "workspace": {
+            "base": dir["base"],
+            "path": config["app"],
+        },
+        "steps": [
+            {
+                "name": "check-suites",
+                "image": OC_CI_ALPINE,
+                "commands": [
+                    "%s/tests/acceptance/check-deleted-suites-in-expected-to-failure.sh" % dir["web"],
```
Comment (saw-jan) [inline .drone.star:1699]: others look good :+1: 
```suggestion
                    "%s/tests/acceptance/check-deleted-suites-in-expected-failure.sh" % dir["web"],
```

**Review by saw-jan — APPROVED** [review]
:+1: :sparkles:

**Pull request merged**